BUG: Fix Series doesn't work in pd.astype(). Now treat Series as dict.

BranYang opened the pull request:
- [x] closes #16717 
 - [x] tests added / passed
 - [x] passes ``git diff upstream/master --name-only -- '*.py' | flake8 --diff``
 - [x] whatsnew entry

BranYang committed:
Make Series works the same as dict in pd.astype()

jreback commented on pandas/tests/frame/test_dtypes.py:
```diff
@@ -487,6 +487,54 @@ def test_astype_dict(self):
         assert_frame_equal(df, original)
 
+    def test_astype_Series(self):
+        # GH16717
```
I would instead modify the previous the test and use parametrize

jreback replied:
then just add the issue number as an additional comment

jreback commented on doc/source/whatsnew/v0.21.0.txt:
```diff
@@ -91,6 +91,8 @@ Bug Fixes
 
 Conversion
 ^^^^^^^^^^
+- Fix a bug when pd.astype() receive Series as `dtype` paramter, no action is taken. Now Series 
```
``Bug in :func:`DataFarme.astype` where a passed Series as a dtype mapping would be ignored``

jreback replied:
Add the issue number as well

BranYang committed:
1. Add test to test Series.astype() as well
2. Use parametrize test instead of redundent another test
3. Add GH # in whatsnew

BranYang committed:
Forget to align changed paramter name.

BranYang committed:
An error in series/test_dtypes.py - a dtype is not used.

BranYang committed:
Change the test method to start with `test_`

BranYang committed:
When `dtype` `is_dict_like` is true, use `get()` instead of `values()` in `pd.Series.astype`

jreback commented on pandas/tests/frame/test_dtypes.py:
```diff
@@ -494,6 +494,13 @@ def test_astype_dict_like(self, dtype_class):
 
+        # if dtypes provided is empty, the resulting DataFrame
+        # should be the same as the original DataFrame
+        dt7 = dtype_class({})
```
add issue number as a comment
use result = 
rather than equiv

jreback commented on doc/source/whatsnew/v0.21.0.txt:
```diff
@@ -92,6 +92,8 @@ Bug Fixes
 Conversion
 ^^^^^^^^^^
+- Fix a bug that nothing happen when `Series` is passed to pd.astype() as `dtype`. Now `Series` 
+works the same as a `dict` when passed to `dtype`. (:issue:`16717`).
```
reword to be less verbose

jreback replied:
you can put this in 0.20.3

jreback commented on pandas/core/generic.py:
```diff
@@ -3505,12 +3505,12 @@ def astype(self, dtype, copy=True, errors='raise', **kwargs):
                     raise KeyError('Only the Series name can be used for '
                                    'the key in Series dtype mappings.')
-                new_type = list(dtype.values())[0]
+                new_type = dtype.get(self.name)
```
``new_type = dtype[self.name]`` is ok here now

BranYang committed:
Move to v0.20.3, and some minor changes.

jreback commented on doc/source/whatsnew/v0.20.3.txt:
```diff
@@ -46,6 +46,7 @@ Conversion
 
 - Bug in pickle compat prior to the v0.20.x series, when ``UTC`` is a timezone in a Series/DataFrame/Index (:issue:`16608`)
 - Bug in Series construction when passing a Series with ``dtype='category'`` (:issue:`16524`).
+- Bug in pd.astype() when passing `Series` as `dtype`. (:issue:`16717`).
```
Bug in ``.astype()`` when passing a ``Series`` to ``dtype`` (use double backticks)

jreback reviewed (approved):
just a whatsnew fix. ping on green.

BranYang committed:
Change the whatsnew.

BranYang commented:
@jreback Done as your suggestion.

jreback committed:
update whatsnew

jreback commented:
thanks!
